editor/src/core/shared/element-path-tree.ts

```diff
@@ -1,140 +1,148 @@
-  }
-
-  return children
-}
-
-function getChildrenPaths(
-  metadata: ElementInstanceMetadataMap,
-  rootPath: ElementPath,
-  paths: ElementPath[],
-): ElementPath[] {
-  const element = metadata[EP.toString(rootPath)]
+    const alreadyAdded = pathString in trees
+    if (!alreadyAdded) {
+      const parentPath = EP.parentPath(path)
+      const subTree = elementPathTree(path, pathString, [], [])
+      trees[pathString] = subTree
+      const pathLastPart = EP.lastElementPathForPath(path)
 
-  // Grab the child elements from the element metadata, if available.
-  let childrenFromElement: ElementPath[] = []
-  if (
-    element != null &&
-    isRight(element.element) &&
-    (isJSXElement(element.element.value) || isJSXFragment(element.element.value)) &&
-    element.element.value.children.length > 0
-  ) {
-    childrenFromElement = element.element.value.children
-      .filter((child) => {
-        if (isFeatureEnabled('Condensed Navigator Entries')) {
-          // if Data Entries are enabled, we should always show them in the Navigator
-          return true
-        } else {
-          return (
-            !isJSXTextBlock(child) &&
-            !isJSExpressionMapOrOtherJavaScript(child) &&
-            !isJSIdentifier(child) &&
-            !isJSPropertyAccess(child) &&
-            !isJSElementAccess(child)
-          )
-        }
-      })
-      .map((child) => EP.appendToPath(rootPath, child.uid))
+      if (pathLastPart?.length === 1) {
+        // If the last part of the path is a single element, it should be added to the inner children.
+        // This would be a path like `a/b/c:d`.
+        trees[EP.toString(parentPath)].innerChildren.push(subTree)
+      } else {
+        // Otherwise this should be added to the props children.
+        trees[EP.toString(parentPath)].propsChildren.push(subTree)
+      }
+    }
   }
 
-  // Then, grab any other child from the paths array, which is not included in the
-  // elements from the metadata.
-  const otherChildrenFromPaths = paths.filter(
-    (path) =>
-      EP.isChildOf(path, rootPath) &&
-      !childrenFromElement.some((other) => EP.pathsEqual(other, path)),
-  )
+  let workingPaths = [...originalPaths]
+  workingPaths.sort((a, b) => {
+    return EP.fullDepth(a) - EP.fullDepth(b)
+  })
+  for (const workingPath of workingPaths) {
+    addPathToTree(workingPath)
+  }
 
-  // If there are children outside the element metadata, need to merge the two and sort them.
-  // Otherwise, return the children from the meta.
-  return otherChildrenFromPaths.length > 0
-    ? childrenFromElement
-        .concat(otherChildrenFromPaths)
-        .sort(
-          (a, b) =>
-            paths.findIndex((p) => EP.pathsEqual(p, a)) -
-            paths.findIndex((p) => EP.pathsEqual(p, b)),
-        )
-    : childrenFromElement
+  for (const elementMetadata of Object.values(metadata)) {
+    forEachRight(elementMetadata.element, (element) => {
+      switch (element.type) {
```

Comment:
Since you only do something when element is jsxelement or jsxfragment, I would just use an `isJSXElementLike` here (without an else)

Reply:
Fixed in 8c91af726.